### PR TITLE
Adding functions documentaion for go docs & codes table in README

### PR DESCRIPTION
**Pull request by escalopa**

### CODE REVIEW

**Comment by lordvidex** [comment]
Hi @escalopa, I'm just noticing this PR. Can you please pull the latest changes I made in the repo, so that I can merge your changes.

**Comment by escalopa** [comment]
Hey @lordvidex, I resolved the merge conflicts into a new branch `docs` on my side, Now this PR is merge from my branch `docs` to  yours `main`.

File: error.go [hunk 1]
```diff
@@ -22,8 +22,9 @@ type Error struct {
 	depth int
 }
 
+// Error returns the error in the format "code: message".
 func (e *Error) Error() string {
-	return e.Code.String() + ": " + strings.Join(e.Msg, ": ")
+	return e.Code.String() + ": " + e.Msg
 }
```
Comment (lordvidex) [inline error.go]: @escalopa `e.Msg` is a list of messages and not just a single message unlike the previous versions.

Comment (lordvidex) [reply error.go]: After this resolution, I will merge

**Pull request merged**